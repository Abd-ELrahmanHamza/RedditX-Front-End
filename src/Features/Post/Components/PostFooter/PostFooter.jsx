import { Container, Comment, MyDropdown } from "./PostFooter.styled";

import { TfiComment } from "react-icons/tfi";
import { VscGift } from "react-icons/vsc";
import { IoMdShareAlt } from "react-icons/io";
import { TbBadge } from "react-icons/tb";
import { HiOutlineDotsHorizontal } from "react-icons/hi";
import { BiHide } from "react-icons/bi";
import { FiFlag } from "react-icons/fi";

import { hidePost } from "Features/Post/Services/postActions";
import useFetchFunction from "Hooks/useFetchFunction";

import { useAuth } from "Features/Authentication/Contexts/Authentication";
import submitSpam from "Features/Post/Services/submitSpam";

/**
 * PostFooter Component that is in the side of Post
 * @returns {React.Component} PostFooter component
 */
const PostFooter = ({ post, setMakeHidden }) => {
  const auth = useAuth();

  /**
   * Custom hook
   */
  const [data, error, isLoading, dataFetch] = useFetchFunction();

  const handleHide = () => {
    setMakeHidden(true);
    hidePost(
      dataFetch,
      {
        linkID: `t3_${post._id}`,
      },
      auth.getToken()
    );
  };
  const handleSpam = () => {
    submitSpam(
      dataFetch,
      {
        linkID: `t3_${post._id}`,
        spamText: "I found that this content is showing violence",
        spamType: "violent content",
      },
      auth
    );
  };
  return (
    <Container>
      <Comment onClick={(event) => {}}>
        <TfiComment /> {post.commentsNumber} Comments
      </Comment>
      <Comment
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <VscGift /> Award
      </Comment>
      <Comment
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <IoMdShareAlt /> Share
      </Comment>
      <Comment
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <TbBadge /> Save
      </Comment>
      <Comment
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <MyDropdown className="d-inline mx-2" autoClose="outside">
          <MyDropdown.Toggle id="dropdown-autoclose-outside">
            <HiOutlineDotsHorizontal />
          </MyDropdown.Toggle>

          <MyDropdown.Menu>
            <MyDropdown.Item href="#">
              <span onClick={handleHide}>
                <BiHide /> Hide
              </span>
            </MyDropdown.Item>
            <MyDropdown.Item href="#">
              <span onClick={handleSpam}>
                <FiFlag /> Report
              </span>
            </MyDropdown.Item>
          </MyDropdown.Menu>
        </MyDropdown>
      </Comment>
    </Container>
  );
};

export default PostFooter;
