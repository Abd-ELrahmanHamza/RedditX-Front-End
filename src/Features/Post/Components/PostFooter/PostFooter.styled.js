import styled from "styled-components";
import Dropdown from "react-bootstrap/Dropdown";

export const Container = styled.div`
  display: flex;
  justify-content: start;
  align-items: center;
  background-color: white;
  padding: 0 20px 20px;
  gap: 2%;
  max-width: 100%;
`;
export const Comment = styled.span`
  color: gray;
  padding: 5px 10px;
  cursor: pointer;
  svg {
    font-size: 24px;
  }

  &:hover {
    background-color: #cccccc2b;
  }
`;
export const MyDropdown = styled(Dropdown)`
 
  .dropdown-menu.show {
    padding: 0;
    overflow: hidden;
    box-shadow: 0px 0px 10px -2px grey;
    
  }

  .dropdown-item {
    
    padding-right: 0!important;
    color: gray;
    display: flex;
    align-items: center;

    span{
        margin: 0;
    }

    &:hover {
      background-color: #5cc0f147;
      color: black !important;
    }
  }

  button {
    background-color: transparent !important;
    border-color: transparent !important;
    color: gray;

    &:after {
      display: none;
    }
  }

  &:hover {
    button {
      background-color: transparent;
      border-color: transparent;
      color: gray;
    }
  }
`;
