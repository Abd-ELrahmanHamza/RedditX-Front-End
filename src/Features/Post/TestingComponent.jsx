// Import contexts providers
import { SubmitDestinationProvider } from "Features/Post/Contexts/submitDestination";
import { CreatePostTitleProvider } from "Features/Post/Contexts/createPostTitle";
import { CreatePostFlagsProvider } from "Features/Post/Contexts/createPostFlags";
import { CreatePostAttachmentsProvider } from "Features/Post/Contexts/createPostAttachments";
import { CreatePostFlairsProvider } from "Features/Post/Contexts/createPostFlairs";
import { CreatePostTextProvider } from "Features/Post/Contexts/createPostText";
import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  const auth = useAuth();
  useEffect(() => {
    auth.login({
      expiresIn: 86400,
      token:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbFR5cGUiOiJiYXJlIGVtYWlsIiwidXNlcm5hbWUiOiJ0Ml9ob21lIiwiaWF0IjoxNjY5OTI0MDEzLCJleHAiOjE2NzAzNTYwMTN9.FDa59vJi5gWy4dQVkEtgyCfm3Pp2BtHkPz0_AfNVKgY",
      username: "t2_home",
    });
  }, []);

  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SubmitDestinationProvider>
          <CreatePostTitleProvider>
            <CreatePostFlagsProvider>
              <CreatePostAttachmentsProvider>
                <CreatePostFlairsProvider>
                  <CreatePostTextProvider>{children}</CreatePostTextProvider>
                </CreatePostFlairsProvider>
              </CreatePostAttachmentsProvider>
            </CreatePostFlagsProvider>
          </CreatePostTitleProvider>
        </SubmitDestinationProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
