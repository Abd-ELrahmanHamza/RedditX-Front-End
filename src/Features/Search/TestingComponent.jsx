// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";
// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { SafeContextProvider } from "./Contexts/SafeSearchContext/Safe-context";
import { SearchContextProvider } from "./Contexts/SearchWordContext/Search-context";

const TestingComponent = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SearchContextProvider>
          <SafeContextProvider>{children}</SafeContextProvider>
        </SearchContextProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default TestingComponent;
