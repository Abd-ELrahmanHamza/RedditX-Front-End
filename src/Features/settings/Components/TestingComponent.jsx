import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        {children}
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
