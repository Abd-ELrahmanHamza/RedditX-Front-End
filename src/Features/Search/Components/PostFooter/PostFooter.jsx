import { Footer } from "./PostFooter.styled";

/**
 * Component that contains the Post Header Footer.
 *
 * @Component
 * @returns {React.Component}
 */

const PostFooter = () => {
  return (
    <Footer>
      <span>15.1k upvotes</span>
      <span>1.7k comments</span>
    </Footer>
  );
};

export default PostFooter;
