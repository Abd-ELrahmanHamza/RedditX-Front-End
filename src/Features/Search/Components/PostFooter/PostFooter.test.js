import PostFooter from "./PostFooter";
import { async } from "@firebase/util";
import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestingComponent from "Features/Search/TestingComponent";
const postfooter = {
  upVotes: 1000,
  Comments: 1000,
};
describe("PostFooter component", () => {
  it("renders check for post Footer", async () => {
    render(
      <TestingComponent>
        <PostFooter postfooter={postfooter} />
      </TestingComponent>
    );

    // const outputElement = screen.getByTitle("commName");
    // expect(outputElement.textContent).toBe("r/communityName");
  });
  it("check for post Footer upVotes Count and Units", async () => {
    render(
      <TestingComponent>
        <PostFooter postfooter={postfooter} />
      </TestingComponent>
    );

    const outputElement = screen.getByTitle("upvote");
    expect(outputElement.textContent).toBe("1k upvotes");
  });
  it("check for post Footer Comments Count and Units", async () => {
    render(
      <TestingComponent>
        <PostFooter postfooter={postfooter} />
      </TestingComponent>
    );

    const outputElement = screen.getByTitle("commentCount");
    expect(outputElement.textContent).toBe("1k comments");
  });
});
// test
