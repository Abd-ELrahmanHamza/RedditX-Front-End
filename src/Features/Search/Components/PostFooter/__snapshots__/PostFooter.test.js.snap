// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Footer this is a test for Post Footer 1`] = `ShallowWrapper {}`;

exports[`Search Page this is a test for Search Page 1`] = `ShallowWrapper {}`;
